src/Mir/TransCustom.hs

```diff
@@ -325,6 +326,36 @@ vector_copy_from_slice = ( ["crucible","vector","{{impl}}", "copy_from_slice"],
     _ -> Nothing
 
 
+-----------------------------------------------------------------------------------------------------
+-- ** Custom: Any
+
+-- Methods for crucible::any::Any (which has custom representation)
+
+any_new :: (ExplodedDefId, CustomRHS)
+any_new = ( ["core", "crucible", "any", "{{impl}}", "new"], \substs -> case substs of
+    Substs [_] -> Just $ CustomOp $ \_ ops -> case ops of
+        [MirExp tpr e] -> do
+            return $ MirExp C.AnyRepr $ R.App $ E.PackAny tpr e
+        _ -> mirFail $ "bad arguments for Any::new: " ++ show ops
+    _ -> Nothing
+    )
+
+any_downcast :: (ExplodedDefId, CustomRHS)
+any_downcast = ( ["core", "crucible", "any", "{{impl}}", "downcast"], \substs -> case substs of
+    Substs [t] -> Just $ CustomOp $ \_ ops -> case ops of
+        [MirExp C.AnyRepr e]
+          | Some tpr <- tyToRepr t -> do
+            let maybeVal = R.App $ E.UnpackAny tpr e
+            let errMsg = R.App $ E.StringLit $ fromString $
+                    "failed to downcast Any as " ++ show tpr
+            let val = R.App $ E.FromJustValue tpr maybeVal errMsg
+            return $ MirExp tpr val
+        _ -> mirFail $ "bad arguments for Any::downcast: " ++ show ops
+    _ -> Nothing
+    )
+
+
+
 -----------------------------------------------------------------------------------------------------
```

Comment:
It would be great if we had a more concise and readable way of writing stuff like this, though it might be that it would take more effort to develop that than it'd be worth.